pkg/collectors/devices/fetcher.go

```diff
@@ -51,9 +51,44 @@ func (inst *fetcher) AddCommand(cmdInst *clients.Cmd) {
 	inst.cmdGrp.AddCommand(cmdInst)
 }
 
+func setValue(field *reflect.StructField, fieldVal reflect.Value, valueName string, value any) error {
```

Comment:
I'm not keen on `valueName` being passed in but then only used for formatting error messages. That suggests to me the error should be simpler from `setValue` then expanded on with the `valueName` at the caller level.